### ISSUE TITLE
[Feat]: Add Calibrated Win Probability & Model Reliability Analysis to Prediction Engine

### ISSUE THREAD

**Issue by OmenSummon** [Arnav-Singh-5080/IPL-Win-Probability-Prediction#33]
## 📌 Problem Statement

CricScope currently displays win probabilities (e.g., **CSK 80% vs MI 20%**) alongside a qualitative confidence tag like **"High"** or **"Low"**. However, these are raw classifier outputs — there is no guarantee that an 80% prediction actually means the team wins ~8 out of 10 times in historically similar match states.

Raw logistic regression probabilities can be systematically overconfident or underconfident depending on the training distribution. Showing uncalibrated numbers as if they are literal real-world likelihoods is statistically misleading to users who trust those percentages at face value.

---

## ✅ Proposed Solution

Add a **probability calibration workflow** to the existing training pipeline and expose the calibrated win probability plus a reliability summary inside the Streamlit dashboard — without changing any existing features or redesigning the UI.

This stays entirely within the current ML stack:
- `scikit-learn`
- Existing Logistic Regression model
- Existing IPL 2008–2020 CSV dataset

---

### 1. Calibration in the Training Pipeline

- Wrap the existing trained model using `scikit-learn`'s `CalibratedClassifierCV`
- Use either:
  - `method='sigmoid'` (Platt Scaling)
  - OR `method='isotonic'`
- Perform calibration on a held-out validation split (e.g., 80/20 train-validation split)
- Keep the existing train/test logic intact
- Save the calibrated model artifact (`.pkl`) alongside or in place of the current model

Example direction:

```python
from sklearn.calibration import CalibratedClassifierCV
```

### 2. Offline Evaluation & Comparison

Add an evaluation script or notebook section that computes and logs the following metrics for both the raw and calibrated models:

| Metric | Purpose |
|---|---|
| Brier Score | Lower score indicates better probability calibration accuracy |
| Log Loss | Penalizes overconfident incorrect predictions |
| Reliability / Calibration Curve | Visual comparison between predicted probabilities and actual win frequencies |

#### Requirements

- Compare the **raw logistic regression model** against the **calibrated model**
- Generate and save a reliability/calibration curve (`.png` or notebook output)
- Document metric comparisons in:
  - The PR description
  - OR a dedicated `calibration_report.md`

---

### 3. Streamlit UI Update

Add a lightweight **"Model Reliability"** card or expander section inside the existing Streamlit prediction dashboard.

#### The section should display:

- Calibrated win probability
- Confidence band or reliability label (Low / Medium / High)
- A short reliability note, for example:

> "Calibrated on IPL 2008–2020 data · Brier Score: 0.18"

#### UI Constraints

- Keep the existing dashboard design intact
- No major UI redesigns
- Match the current Streamlit styling/theme
- The reliability section should feel native to the existing interface

---

I would like to work on this issue under Gssoc26 @Arnav-Singh-5080

**Comment by shreya975** [comment]
Hi @Arnav-Singh-5080,

I’d like to work on this issue.

I’ve gone through the current prediction pipeline and this feature can significantly improve the trustworthiness of the win probability engine by introducing probability calibration and reliability evaluation.

My planned approach is:

* analyze current prediction confidence behavior
* implement probability calibration techniques (Platt Scaling / Isotonic Regression based on model compatibility)
* add reliability analysis metrics such as Brier Score and calibration curves
* visualize confidence vs actual outcome alignment
* ensure the calibrated probabilities integrate cleanly with the existing prediction workflow and dashboards

I also plan to keep the implementation modular so additional models can later reuse the same calibration pipeline.

I have prior experience working on ML-focused open source contributions and would love to contribute this enhancement properly.

**Comment by OmenSummon** [comment]
Thank you @Arnav-Singh-5080 i will start working in it as soon as possible I have 3 End sem exams this week.
The issue will be cleared by 23rd may

**Comment by shreya975** [comment]
Hi @Arnav-Singh-5080,

I have completed the implementation for this issue ✅

### Implemented
- Probability calibration using `CalibratedClassifierCV`
- Reliability analysis with Brier Score & Log Loss
- Confidence labels (High / Medium / Low)
- Calibrated win probability display in the Streamlit UI
- Win probability progression visualization

Looking forward to your review.
